tripal/tripal
# Add formatter settings to fields that need them

# Tripal 4 Core Dev Task

### Closes #1692

### Tripal Version: 4.x

## Description
This PR adds field display settings to:
 * ChadoBooleanTypeFormatter - configure the value displayed for each of true and false, _e.g._ "Yes" and "No"
 * ChadoSequenceLengthFormatterDefault - configurable prefix and suffix, _e.g._ suffix of " b.p.", as well as thousand separator character _e.g._ ",", ".", " "
 * ChadoSequenceFormatterDefault - configurable line length and autosizing window up to container width, defaults to 50 as before
 * ChadoSequenceChecksumFormatterDefault - configure either upper case, lower case, or as stored in the database table.
 * `ChadoSequenceLengthFormatterCommas.php` is deleted as it is now no longer needed (sorry @srobb1 )

## Testing?
 * Create a gene record for testing, with a sequence at least 1,000 b.p. long
 * Go to the gene display settings page, observe that with this PR there are now settings cogs.
![settings_page](https://github.com/tripal/tripal/assets/8419404/1cb0b9a7-c34c-43a4-9413-b84e14d756f7)

 * For the "Sequence Length" field, enter a suffix value, " b.p." is a logical choice, and a thousand separator character such as "," or "." or " " (space)
![integer_type_settings](https://github.com/tripal/tripal/assets/8419404/f4545fb9-5690-43f5-b786-9c987035604b)

 * For the "Sequence Residues" field, change wrap value to "100". Validation prevents a negative number, or a non-number. (Default is 50)
![residues](https://github.com/tripal/tripal/assets/8419404/8b87ac16-11ba-4db9-8003-57f9c1b3705e)

 * For the "Sequence MD5 Checksum" field, select either upper case or lower case. (Default is "As stored")
![md5](https://github.com/tripal/tripal/assets/8419404/b9bdbf7a-c879-4bf2-935a-89f5a4873cf3)
 
 * For the "Is Analysis" or "Is Obsolete" fields, enter new text for true and false. The validation here is they cannot be blank, and they cannot both be the same. (Default is "True" and "False")
![boolean](https://github.com/tripal/tripal/assets/8419404/87de9a75-c94d-45c4-941f-e85efb4a703b)

 * Observe the gene page. For example:
![gene_example_1](https://github.com/tripal/tripal/assets/8419404/2d953528-8431-434e-bbb3-6aa79988c3a1)
